p4-16/psa/PSA.mdk

```diff
@@ -1748,3 +1748,357 @@ types, as the P4 program author wishes.
+|--------------|-------------|---------------|--------------------|
+~
+
+
+Table [#results-of-one-pkt-thru-ingress-or-egress] shows what can
+happen to a packet as a result of a single time being processed in
+ingress, or a single time being processed in egress.  The cases are
+the same as shown in Table [#packet-path-abbreviations], but have been
+grouped together by similar values of "Processed next by".
+
+
+~ TableFigure { #results-of-one-pkt-thru-ingress-or-egress; \
+  caption: Result of packet processed one time by ingress or egress.}
+|--------------|-------------|---------------|--------------------|
+|              |             | Processed     | Resulting          |
+| Abbreviation | Description | next by       | packet(s)          |
++:-------------+:------------+:--------------+:-------------------+
+| NFP          | normal packet from port |         | At most one CI2E |
+|--------------|-------------|           |                    |
+| NFCPU        | packet from CPU port    | ingress | packet, plus at most |
+|--------------|-------------|           |                    |
+| RESUB        | resubmitted packet      |         | one of a RESUB,  |
+|--------------|-------------|           |                    |
+| RECIRC       | recirculated packet     |         | NU, or NM packet |
+|--------------|-------------|-----------|--------------------|
+| NU           | normal unicast packet       |         | At most one CE2E     |
+|              | from ingress to egress      |         | packet, plus at most |
+|--------------|-------------|               |                    |
+| NM           | normal multicast-replicated | egress  | one of a RECIRC,     |
+|              | packet from ingress to egress |       | NTP, or NTCPU packet |
+|--------------|-------------|               |                    |
+| CI2E         | clone from ingress to egress |        |          |
+|--------------|-------------|               |                    |
+| CE2E         | clone from egress to egress |         |          |
+|--------------|-------------|---------------|--------------------|
+| NTP          | normal packet to port   | device at other | determined by the |
+|              |                         | end of port     | other device      |
+|--------------|-------------|---------------|--------------------|
+| NTCPU        | normal packet to CPU port | CPU | determined by CPU |
+|--------------|-------------|---------------|--------------------|
+~
+
+For egress packets, the choice of egress port, or the port to the CPU,
+is made by the immediately previous processing step (ingress for NU,
+NM, or CI2E packets, egress for CE2E packets), and egress processing
+cannot change that choice to a different port.  It can change the
+packet to RECIRC instead, or drop the packet.
+
+A single packet received by a PSA system from a port can result in 0,
+1, or many packets going out, all under control of the PSA program.
```

Comment:
Nit: P4 program

Reply:
I think I wrote "PSA program" after noticing at least one other place in the document that uses that phrase.  In some cases, when saying that something is or is not allowed in a PSA program, it seems more precise than saying P4 program -- we don't want the PSA spec to limit what programs people expect to be able to write in other architectures.

I'm fine either way, and we can do a global search and replace "PSA program" -> "P4 program" if you prefer.

Reply:
Made the replacements in the new text of this PR.

Reply:
I think we should use P4 program and PSA implementation consistently throughout the document. If we agree, we can have these changes done during the editorial phase as well.